crates/block-producer/src/server/mod.rs

```diff
@@ -56,9 +55,11 @@ pub async fn serve(config: BlockProducerConfig) -> Result<()> {
 
     let addr = config
         .endpoint
-        .to_socket_addrs()?
+        .to_socket_addrs()
+        .expect("Failed to turn address into socket address.")
```

Comment:
Please, avoid panics in the library code. Consider adding new error enumeration, if this function can return different error types.